### DISCUSSION TITLE
让你的SSH通过HTTP代理或者SOCKS5代理

### DISCUSSION THREAD

**Discussion by nagaame** [nagaame/giscus-discussions#6] [Announcements]
# 让你的SSH通过HTTP代理或者SOCKS5代理

socks5代理 有时候你的SSH因为某些原因, 或者你的VPS的SSH端口被某些无法阻挡的力量被封禁了, 你使用普通的SSH方式无法登录你的服务器, 所以这时候我们需要使用一些方式, 让你的SSH通过代理链接你的服务器, 进而解决这个问题.  直接配置 如何让你的ssh链接使用socks代理呢, 我们可以使用nc命令, 使用ssh的ProxyCommand配合nc可以让ssh通过你设置的代理访问服务器  $ ssh -o ProxyCommand=&quot;nc -X 5 -x 127.0.0.1:1080 %h %p&quot; root@server 其中使用ProxyCommand命令, 带上具体内容nc -X 5 -x 127.0.0.1:1080 %h %p, 127.0.0.1:1080是你的代理实际地址和端口 最后边的root@server是你需要登录的服务器和用户名  nc命令的常用参数:  -X是指定代理协议

https://kuokuo.io/2019/07/01/ssh-over-http-or-socks/

**Comment by anztim** [comment]
windows下没有nc，有什么替代方法吗

**Comment by nagaame** [comment]
> @anztim
> windows下没有nc，有什么替代方法吗

可以尝试看这里, 我很少用windows, 具体情况就不是很清楚了, 应该需要用到mingw
链接: https://blog.imyan.ren/posts/4d70a82d/

**Comment by lijinma** [comment]
nc 也可以实现http代理

**Comment by iFence** [comment]
使用socks5，需要在服务器上先安装对吧

**Reply by nagaame** [reply]
是的

**Comment by Lotfree618** [comment]
你好，请问这样报错大概是什么问题呢？

 ssh -o ProxyCommand="nc -x 127.0.0.1:7777 %h %p" lotfree@10.108.3.50 -v                                        
OpenSSH_9.2p1 Debian-2, OpenSSL 3.0.9 30 May 2023
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: include /etc/ssh/ssh_config.d/*.conf matched no files
debug1: /etc/ssh/ssh_config line 21: Applying options for *
debug1: Executing proxy command: exec nc -x 127.0.0.1:7777 10.108.3.50 22
debug1: identity file /home/lotfree/.ssh/id_rsa type 0
debug1: identity file /home/lotfree/.ssh/id_rsa-cert type -1
debug1: identity file /home/lotfree/.ssh/id_ecdsa type -1
debug1: identity file /home/lotfree/.ssh/id_ecdsa-cert type -1
debug1: identity file /home/lotfree/.ssh/id_ecdsa_sk type -1
debug1: identity file /home/lotfree/.ssh/id_ecdsa_sk-cert type -1
debug1: identity file /home/lotfree/.ssh/id_ed25519 type -1
debug1: identity file /home/lotfree/.ssh/id_ed25519-cert type -1
debug1: identity file /home/lotfree/.ssh/id_ed25519_sk type -1
debug1: identity file /home/lotfree/.ssh/id_ed25519_sk-cert type -1
debug1: identity file /home/lotfree/.ssh/id_xmss type -1
debug1: identity file /home/lotfree/.ssh/id_xmss-cert type -1
debug1: identity file /home/lotfree/.ssh/id_dsa type -1
debug1: identity file /home/lotfree/.ssh/id_dsa-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_9.2p1 Debian-2
kex_exchange_identification: Connection closed by remote host
Connection closed by UNKNOWN port 65535

**Reply by nagaame** [reply]
这个错误比较棘手, 你可以看看[这里](https://stackoverflow.com/questions/72430560/connection-closed-by-unknown-port-65535-when-attempting-ssh-in-corporate-proxy) 或者尝试不加代理的情况下 ssh登录到服务器试试, 还是不行的话, 只能通过其他方式连接服务器了, 比如说web方式: https://github.com/tsl0922/ttyd

**Reply by Lotfree618** [reply]
谢谢，这是因为我们学校的校园网屏蔽了ssh，导致我不得不想办法，我发现http和socks连接并没有屏蔽我才想这样的办法，谢谢你的博客对我参考很大
